File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/annos/ErdRolesAllowed.java
package org.javamaster.invocationlab.admin.annos;

import org.javamaster.invocationlab.admin.enums.RoleEnum;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.METHOD;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

/**
 * @author yudong
 */
@Documented
@Retention(RUNTIME)
@Target({METHOD})
public @interface ErdRolesAllowed {
    RoleEnum[] value();

    String msg() default "无权操作";
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/annos/ErdRolesAllowed.kt
package org.javamaster.invocationlab.admin.annos

import org.javamaster.invocationlab.admin.enums.RoleEnum

/**
 * @author yudong
 */
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class ErdRolesAllowed(vararg val value: RoleEnum, val msg: String = "无权操作")
